graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -173,6 +174,32 @@ public UserSummary get(@ApiParam(name = "username", value = "The username to ret
         return toUserResponse(user, isSelf || canEditUserPermissions, AllUserSessions.create(sessionService));
     }
 
+    @GET
+    @Path("id/{userId}")
+    @ApiOperation(value = "Get user details by userId", notes = "The user's permissions are only included if a user asks for his " +
+            "own account or for users with the necessary permissions to edit permissions.")
+    @ApiResponses({
+            @ApiResponse(code = 404, message = "The user could not be found.")
+    })
+    public UserSummary getbyId(@ApiParam(name = "userId", value = "The userId to return information for.", required = true)
+                           @PathParam("userId") String userId,
+                           @Context UserContext userContext) {
+
+        final User user = loadUserById(userId);
+        final String username = user.getName();
+        // If a user has permissions to edit another user's profile, it should be able to see it.
```

Comment:
How about putting the handling code into a separate method so we don't duplicate the same code in `get` and `getbyId`? :smile: